### PR TITLE
GT-176 Add support for Inverted index 

### PR DESCRIPTION
**Pull request by jwierzbo**

### CODE REVIEW

**Comment by nikita-vanyasin** [comment]
Build is failing due to the fact new field name (`compression`) is not on 3.10-nightly build yet.

File: collection_indexes_impl.go [hunk 1]
```diff
@@ -313,6 +313,46 @@ func (c *collection) EnsureZKDIndex(ctx context.Context, fields []string, option
...
 
+type invertedIndexData struct {
+	InvertedIndexOptions
+	Type string `json:"type"`
+	ID   string `json:"id,omitempty"`
+
+	ArangoError `json:",inline"`
+}
+
+func (c *collection) EnsureInvertedIndex(ctx context.Context, options *InvertedIndexOptions) (Index, bool, error) {
```
[the remark this answers is not in the document]
Comment (nikita-vanyasin) [reply collection_indexes_impl.go:326]: done

File: collection_indexes.go [hunk 2]
```diff
@@ -196,3 +200,52 @@ type EnsureZKDIndexOptions struct {
...
+	// IncludeAllFields If set to true, all fields of this element will be indexed. Defaults to false.
+	IncludeAllFields bool `json:"includeAllFields,omitempty"`
+	// TrackListPositions If set to true, values in a listed are treated as separate values. Defaults to false.
+	TrackListPositions bool `json:"trackListPositions,omitempty"`
+	// Fields contains the properties for individual fields of the element.
+	// The key of the map are field names.
+	Fields []InvertedIndexField `json:"fields,omitempty"`
+}
+
+type InvertedIndexPrimarySort struct {
```
[the remark this answers is not in the document]
Comment (nikita-vanyasin) [reply collection_indexes.go:232]: done

File: collection_indexes.go [hunk 3]
```diff
@@ -196,3 +200,52 @@ type EnsureZKDIndexOptions struct {
...
+	Fields []ArangoSearchPrimarySortEntry `json:"fields,omitempty"`
+	// Compression optional
+	Compression PrimarySortCompression `json:"compression,omitempty"`
+}
+
+type InvertedIndexField struct {
```
[the remark this answers is not in the document]
Comment (nikita-vanyasin) [reply collection_indexes.go:239]: done

**Review by informalict — APPROVED** [review]
LGTM

**Pull request merged**